tests/unit/test_column_selector.py

```diff
@@ -37,6 +37,20 @@ def test_constructor_works_with_single_subgroups_and_lists():
     assert selector2.subgroups[0] == ColumnSelector(["a", "b", "c"])
 
 
+def test_constructor_too_many_level():
+
+    selector0 = ColumnSelector([], subgroups=ColumnSelector(["a", "b", "c"]))
+    selector1 = ColumnSelector(["f", "w", "g"], subgroups=selector0)
+    # next should break
+
+    test_pass = False
```

Comment:
To test that a method raises an error, you can do `with pytest.raises(AttributeError) as exc_info:` as a context manager. The test will fail if the code inside the context block doesn't raise the error, and you can make assertions against the error message (e.g. to check the error message) after the block.

Reply:
(Here's [an example](https://github.com/NVIDIA/NVTabular/pull/1014/files#diff-8f19358af314f63dd2835237d2bcbf90c9065980912f10c2f77ca98d990ba746R40-R51) from the `ColumnSelector` PR. I have to look up how to do this every time I want to do it.)